src/components/ui/Checkbox.tsx

```diff
@@ -0,0 +1,43 @@
+'use client';
+
+/**
+ * This file is based on the Checkbox component from shadcn/ui and customized for our needs.
+ *  * The Checkbox component of shadcn/ui is based on  Checkbox component of Radix UI.
+ *
+ * See the official docs for more info:
+ * shadcn/ui: https://ui.shadcn.com/docs/components/checkbox
+ *  * Radix UI: https://www.radix-ui.com/themes/docs/components/checkbox
+ *
+ */
+
+import { CheckIcon } from '@/assets/images/icons';
+import { cn } from '@/lib/tailwind/utils';
+
+import { Indicator as CheckboxIndicator, Root as CheckboxRoot } from '@radix-ui/react-checkbox';
+import { ComponentPropsWithoutRef, ElementRef, forwardRef } from 'react';
+
+import { Icon } from '.';
+
+const Checkbox = forwardRef<
+  ElementRef<typeof CheckboxRoot>,
+  ComponentPropsWithoutRef<typeof CheckboxRoot>
+>(({ className, ...props }, ref) => (
+  <CheckboxRoot
+    ref={ref}
+    className={cn(
+      'peer',
+      'h-6 w-6 shrink-0 rounded-sm border border-primary',
+      'disabled:cursor-not-allowed',
```

Comment:
Do you think it's better to add some typical style for disabled state?
Since it's currently not designed in wire frame, I removed `disabled:cursor-not-allowed` in the latest commit!

Reply:
Thank you for updating! 

In general, disabled checkboxes are usually styled in a design. However, since our current features do not require disabled checkboxes, I suppose there is no need to define a style for them for now.